packages/theme/middleware/checkout.js

```diff
@@ -1,35 +1,58 @@
-const cartEmpty = cart => cart.itemCount === 0;
-const canEnterBilling = cart => Boolean(cart.address?.shipping) && cart.lineItems.length;
-const canEnterPayment = cart => Boolean(cart.address?.billing) && Boolean(cart.address?.shipping) && cart.lineItems.length;
+import { Logger } from '@vue-storefront/core';
 
 export default async ({ app, $vsf }) => {
   const currentPath = app.context.route.fullPath.split('/checkout/')[1];
-
   if (!currentPath) return;
 
-  const cart = await $vsf.$spree.api.getCart();
+  const tryToGetShipments = async () => {
+    try {
+      return await $vsf.$spree.api.getShipments();
+    } catch (err) {
+      Logger.error('checkout-middleware/tryToGetShipments', err);
+    }
+    return null;
+  };
+
+  const getCart = async () => {
+    return await $vsf.$spree.api.getCart();
+  };
 
   const redirectWithLocalePath = (path) => {
     app.context.redirect(app.localePath(path));
   };
 
+  const checks = await (async () => {
```

Comment:
Maybe we should extract the internal async function to a named constant, e.g. getChecks or computeChecks?
It will make it a bit easier to read.

Reply:
We have an initiation code that we don't need to use again (shipments, cart), so it's a valid use case to use the IIFE pattern. IMHO, as we will not reuse the code again, using IIFE, in this case, is better than using a function declaration or a function expression, because it doesn't pollute the module namespace with unnecessary information.

I wanted to include this checks initiation inside of the middleware function to gain direct access to $vsf and `tryToGetShipments` and `getCart` functions.

What do you think?